AsyncDisplayKit/Details/ASDataController+Subclasses.h

```diff
@@ -0,0 +1,99 @@
+/* Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "ASDataController.h"
+
+@interface ASDataController (Subclasses)
+
+/**
+ * Queues the given operation until an `endUpdates` synchronize update is completed.
+ *
+ * If this method is called outside of a begin/endUpdates batch update, the block is
+ * executed immediately.
+ */
+- (void)performEditCommandWithBlock:(void (^)(void))block;
+
+/**
+ * Safely locks access to the data source and executes the given block, unlocking once complete.
+ *
+ * When `asyncDataFetching` is enabled, the block is executed on a background thread.
+ */
+- (void)accessDataSourceWithBlock:(dispatch_block_t)block;
+
+- (void)willPerformInitialDataLoading;
+
+/**
+ * An opportunity for a subclass to access the data source before entering into the editing queue
+ */
+- (void)prepareForReloadData;
```

Comment:
This seems vague - is it specific to reloadData, or any edit operation?  If any edit operation, it should be renamed.  If specific to reloadData, the comment seems too broad and should be edited to clarify it is only related to reload data.